### ISSUE TITLE
no-identical-title regression after update from 10.5.0 -> 11.2.0

### ISSUE THREAD

**Issue by atombrenner** [lo1tuma/eslint-plugin-mocha#407]
Hi, 
after updating from `10.5.0` to `11.2.0` many `mocha/no-identical-title` problems were reported. 
Regarding to the [documentation](https://github.com/lo1tuma/eslint-plugin-mocha/blob/main/docs/rules/no-identical-title.md) two test cases in different describe blocks should not raise the problem:

```typescript

import { describe } from 'mocha' 

describe('test suite', () => {
  describe('a context', () => {
    it('throws an exception', () => {});
  });

  describe('another context', () => {
    it('throws an exception', () => {}); // should be OK, but gives error
  });
});
```

**Comment by atombrenner** [comment]
I am using eslint version `9.39.2`

**Comment by atombrenner** [comment]
I looked into the source of this plugin and tried to create a test case in `no-identical-title.test.ts` that reproduces the problem so that I can work on a fix. To my surprise the test case succeeded. Then I realized that I use a typescript project and the problem might be related to the parser configured by [typescript-eslint](https://typescript-eslint.io/).
Then I created a new project to reproduce the problem with minimal context, again without success. I was not even able to get mocha eslint rules to fire on purpose. 

And then I realized the real root cause. It seems, if you explicitly import `describe` or `it` from `mocha`, then tests are not checked at all! So the full examples was:

```typescript
import { expect } from '../testingHelper';
import { describe } from 'mocha' // `describe` is imported, but `it` was forgotten by original author

describe('test suite', () => {
  describe('one context', () => {
    it('throws an exception', () => {
```

In my fresh test project I imported `describe` and `it` so no mocha eslint rule was checked. In my real legacy project, sometimes only `describe` was imported, so the mocha eslint plugin saw only the `it` test cases and that caused the duplicate title error. 🤦 😃 
(Updated the my first post accordingly)

**Comment by atombrenner** [comment]
Why are tests not checked that use imported `describe` or `it` functions instead of using globals? 
In my huge legacy code base multiple authors used different styles over the years, no imports or mixed imports (probably because of auto import or copilot). After the upgrade to v11 only test without any imports are checked.
How can I ensure, that tests are checked if authors in the future by mistace add those problematik import statements?
